### ISSUE TITLE
core(dialog): dialog cannot be labelled automatically (from the template)

### ISSUE THREAD

**Issue by platon-rov** [SAP/fundamental-ngx#6052]
#### Is this a bug, enhancement, or feature request?

Enhancement, bug.

#### Briefly describe your proposal.

As dialogs might be created from the code it would be great to have the ability to automatically label the dialog as it's not possible to specify `ariaLabelledBy` in the config at the moment of creation.

Example:

We have `TableP13DialogComponent` component which creates different dialogs, for example `P13SortingDialogComponent`. 

As we want `aria-labelledby` attribute for every `P13SortingDialogComponent` instance (and any other dialogs) we should create a counter in the `TableP13DialogComponent` component, update the counter when any sort (and others) dialogs are created, generate and pass the ID (looks like `sort-dialog-${id}`) to the dialog config and to the `P13SortingDialogComponent` to reference created ID with the title in the template.

We should repeat these steps for any similar cases.

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

Latest.

#### If this is a bug, please provide steps for reproducing it.

.

#### Please provide relevant source code if applicable.

.

#### Is there anything else we should know?

[Example](https://material.angular.io/components/dialog/overview) from Angular Material.
